### PR TITLE
Add group.protocol config handling for streams

### PR DESCRIPTION
**Pull request by lucasbru**
- accept new streams group.protocol value
 - do not forward to consumer clients
 - log message that not for production

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java [hunk 1]
```diff
@@ -841,6 +841,8 @@ public class StreamsConfig extends AbstractConfig {
     public static final String WINDOW_STORE_CHANGE_LOG_ADDITIONAL_RETENTION_MS_CONFIG = "windowstore.changelog.additional.retention.ms";
     private static final String WINDOW_STORE_CHANGE_LOG_ADDITIONAL_RETENTION_MS_DOC = "Added to a windows maintainMs to ensure data is not deleted from the log prematurely. Allows for clock drift. Default is 1 day";
 
+    private static final String[] NOT_CONFIGURABLE_IN_GLOBAL_CONSUMER =
+        new String[] {ConsumerConfig.GROUP_PROTOCOL_CONFIG};
```
Comment (cadonna) [inline streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java]: Why is this needed for the global consumer but not for the restore consumer? Both are manually assigned.

Comment (cadonna) [reply streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java]: Ah, I think I misunderstood. You did not mean the global consumer here, did you?

**Review by cadonna — COMMENTED** [review]
Thanks @lucasbru !

I have a question.

File: streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java (hunk 1 above)
Comment (cadonna) [reply streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java]: If I indeed misunderstood, you should choose a different name.

Comment (lucasbru) [reply streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java]: yeah it was called "global" in another place, I was also wondering about the name. I agree global is confusing.

Howabout "ROOT"? the root of the property tree

Comment (cadonna) [reply streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java]: What about `UNSETTABLE_COMMON_CONSUMER_CONFIGS`?

Comment (cadonna) [reply streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java]: Or `NON_SETABLE_COMMON_CONSUMER_CONFIGS`?

Comment (lucasbru) [reply streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java]: not sure. all the other static use "NOT_CONFIGURABLE", why use "UNSETTABLE" or "NON_SETABLE", is it better?

Let me suggest `CONSUMER_CONFIGS_IGNORED_IN_ROOT_SCOPE`.

Comment (cadonna) [reply streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java]: IMO, `ROOT_SCOPE` is really hard to understand. It does also not follow the `NOT_CONFIGURABLE` (actually it should be `NON_CONFIGURABLE`).

But I like `IGNORED`. What about `IGNORED_COMMON_CONSUMER_CONFIGS`?

Comment (lucasbru) [reply streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java]: It changes the meaning. as far as I understand this class, common consumer config are defined in the `consumer` prefix, so we'd be talking about. `consumer.group.protocol`, but what we are ignoring is just `group.protocol`. So I think this would be incorrect

Comment (cadonna) [reply streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java]: I understood now, what you mean. This is quite confusing, because the Streams config shadows the consumer config. Although, the configs have the same name, they are used in different places. We do not want to pass `group.protocol` to the consumer. Other unprefixed consumer configs we want to pass into the consumer. 🤯 
What about `IGNORED_UNPREFIXED_CONSUMER_CONFIGS`?
That is my last suggestion. Take it or use `CONSUMER_CONFIGS_IGNORED_IN_ROOT_SCOPE`.

Comment (lucasbru) [reply streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java]: sure, lets use unprefixed
